R/estimateParams.R

```diff
@@ -53,3 +53,366 @@ prepareData <- function(data) {
+#' @export
+#'
+#' @details
+#' Extracts initial parameter values specifically for an exponential transition model.
+#'
+#' @examples
+#' transition <- exponential_transition(h01 = 1.2, h02 = 1.5, h12 = 1.6)
+#' getInit(transition)
+getInit.ExponentialTransition <- function(transition) {
+  unlist(transition$hazards, use.names = FALSE)
+}
+
+#' Retrieve Initial Parameters for Weibull Transition Model
+#'
+#' @param transition (`WeibullTransition`)\cr containing the initial parameters.
+#'   See [weibull_transition()] for details.
+#'
+#' @return Returns a vector (`numeric`) of initial parameters for the Weibull transition.
+#' @export
+#'
+#' @details
+#' Extracts initial parameter values specifically for a Weibull transition model.
+#'
+#' @examples
+#' transition <- weibull_transition(h01 = 1.2, h02 = 1.5, h12 = 1.6, p01 = 2, p02 = 2.5, p12 = 3)
+#' getInit(transition)
+getInit.WeibullTransition <- function(transition) {
+  c(unlist(transition$hazards, use.names = FALSE), unlist(transition$weibull_rates, use.names = FALSE))
+}
+
+#' Generate the Target Function for Optimization
+#'
+#' @param params (`numeric` vector)\cr
+#'   Vector of parameters to be optimized over.
+#' @param data (`data.frame`)\cr
+#'   Data frame in the format created by [prepareData()].
+#' @param transition (`transitionParameters` object)\cr
+#'   Object specifying the distribution family. See [exponential_transition()] or [weibull_transition()] for details.
+#'
+#' @return Returns a function that calculates the negative log-likelihood (`numeric`) for the given parameters.
+#' @export
+#'
+#' @details
+#' This function creates a target function for optimization, which computes the negative log-likelihood for given
+#' parameters, data, and transition model type.
+#'
+#' @examples
+#' transition <- exponential_transition(2, 1.3, 0.8)
+#' simData <- getOneClinicalTrial(
+#'   nPat = c(30), transitionByArm = list(transition),
+#'   dropout = list(rate = 0.8, time = 12),
+#'   accrual = list(param = "time", value = 1)
+#' )
+#' params <- c(1.2, 1.5, 1.6) # For ExponentialTransition
+#' data <- prepareData(simData)
+#' transition <- exponential_transition()
+#' getTarget(params, data, transition)
+getTarget <- function(params, data, transition) {
+  assert_numeric(params)
+  assert_data_frame(data)
+  assert_class(transition, "TransitionParameters")
+  if ("ExponentialTransition" %in% class(transition)) {
+    negLogLik(transition = exponential_transition(h01 = params[1], h02 = params[2], h12 = params[3]), data = data)
+  } else {
+    negLogLik(transition = weibull_transition(
+      h01 = params[1], h02 = params[2], h12 = params[3],
+      p01 = params[4], p02 = params[5], p12 = params[6]
+    ), data = data)
+  }
+}
+
+#' Format Results of Parameter Estimation for Different Transition Models
+#'
+#' @param transition (`ExponentialTransition` or `WeibullTransition`)\cr
+#'   See [exponential_transition()] or [weibull_transition()] for details.
```

Comment:
it would be easier to always just use the superclass here in the argument, so 
```suggestion
#' @param transition (`TransitionParameters`)\cr
#'   see [exponential_transition()] or [weibull_transition()] for details.
```